src/components/insights/cards/__tests__/CardDashboard.spec.js

```diff
@@ -0,0 +1,126 @@
+import { beforeEach, describe, expect, it } from 'vitest';
+import { mount } from '@vue/test-utils';
+
+import CardDashboard from '../CardDashboard.vue';
+import CardTitleError from '../CardTitleError.vue';
+import IconLoading from '@/components/IconLoading.vue';
+
+const emojis = {
+  smile: { skins: [{ native: '😄' }] },
+};
+
+const createWraper = (props) => {
+  return mount(CardDashboard, {
+    props: { metric: '100', configurable: true, configured: true, ...props },
+    global: {
+      stubs: { CardTitleError: true, IconLoading: true },
+      mocks: { emojis },
+    },
+  });
+};
+
+describe('CardDashboard', () => {
+  let wrapper;
+  beforeEach(() => {
+    wrapper = createWraper({});
+  });
+
+  it('should render the CardTitleError component when showMetricError is true', async () => {
+    await wrapper.setProps({ metric: null });
+    expect(wrapper.findComponent(CardTitleError).exists()).toBe(true);
+  });
+
+  it('should not render the CardTitleError component when showMetricError is false', async () => {
+    expect(wrapper.findComponent(CardTitleError).exists()).toBe(false);
+  });
+
+  it('should render the loading icon when isLoading is true', async () => {
+    await wrapper.setProps({ isLoading: true });
+    expect(wrapper.findComponent(IconLoading).exists()).toBe(true);
```

Comment:
In addition to checking IconLoading, is there any component/text/icon that ceases to exist when it is true? If so, it might be interesting to check